scripts/check.sh

```diff
@@ -87,67 +95,96 @@ while [[ $# -gt 0 ]]; do
       ;;
 
     --allow-dirty)
-      allow_dirty=true
+      ALLOW_DIRTY=true
+      ;;
+
+    --amend)
+      COMMIT_METHOD=amend
+      ;;
+
+    --fixup)
+      COMMIT_METHOD=fixup
       ;;
 
     --commit)
-      commit=true
+      COMMIT_METHOD=message
       ;;
 
     *)
       if git rev-parse "$1" >& /dev/null; then
-        start="$1"
+        START_REVISION="$1"
         break
       fi
       ;;
   esac
   shift
 done
 
-if [[ $allow_dirty == true && $commit == true ]]; then
-  echo "--allow_dirty and --commit are mutually exclusive"
+if [[ "${ALLOW_DIRTY}" == true && "${COMMIT_METHOD}" == "message" ]]; then
+  echo "--allow-dirty and --commit are mutually exclusive"
   exit 1
 fi
 
 if ! git diff-index --quiet HEAD --; then
-  if [[ $allow_dirty != true ]]; then
+  if [[ "${ALLOW_DIRTY}" != true ]]; then
     echo "You have local changes that could be overwritten by this script."
     echo "Please commit your changes first or pass --allow-dirty."
     exit 2
   fi
 fi
 
 # Record actual start
-start_sha=$(git rev-parse "$start")
+START_SHA=$(git rev-parse "${START_REVISION}")
+HEAD_SHA=$(git rev-parse HEAD)
+
+function maybe_commit() {
+  local message="$1"
+
+  if [[ "${COMMIT_METHOD}" == "none" ]]; then
+    return
+  fi
+
+  echo "${message}"
+  case "${COMMIT_METHOD}" in
+    amend)
+      git commit -a --amend -C "${HEAD_SHA}"
+      ;;
+
+    fixup)
+      git commit -a --fixup="${HEAD_SHA}"
+      ;;
+
+    message)
+      git commit -a -m "${message}"
+      ;;
+
+    *)
+      echo "Unknown commit method ${COMMIT_METHOD}" 1>&2
+      exit 2
+      ;;
+  esac
+}
 
 # Restyle and commit any changes
-scripts/style.sh "$start_sha" || :
+"${top_dir}/scripts/style.sh" "${START_SHA}"
 if ! git diff --quiet; then
-  if [[ $commit == true ]]; then
-    echo "Style generated changes"
-    git commit -a --fixup=HEAD
-  fi
+  maybe_commit "style.sh generated changes"
 fi
 
 # If there are changes to the Firestore project, ensure they're ordered
 # correctly to minimize conflicts.
-if ! git diff --quiet --name-only --diff-filter=ACMR -- \
-    Firestore/Example/Firestore.xcodeproj; then
-
-  scripts/sync_project.rb
+if ! git diff --quiet -- Firestore/Example/Firestore.xcodeproj; then
+  "${top_dir}/scripts/sync_project.rb"
   if ! git diff --quiet; then
-    if [[ $commit == true ]]; then
-      echo "Sync Xcode project"
-      git commit -a --fixup=HEAD
-    fi
+    maybe_commit "sync_project.rb generated changes"
   fi
 fi
 
-# Check lint errors
-(
-  scripts/lint.sh "$start_sha"
-  scripts/check_copyright.sh
-  scripts/check_no_module_imports.sh
-  scripts/check_test_inclusion.py
-  scripts/check_whitespace.sh
-) 2>&1 | sed "s,^\\([A-Za-z]*/\\),$top_dir/\\1,"
+# Check lint errors.
+#
+# Keep in sync with the checks stage in .travis.yml
```

Comment:
You could refactor this: Have both .travis.yml and this script call `./scripts/travis_checks.sh $commit`. You'd end up duplicating the style.sh check... but I'm not sure that matters.

Optional.

Reply:
I'll have travis just call this script.